### ISSUE TITLE
Files Resource Picker Api

### ISSUE THREAD

**Issue by MathewsJoseph25** [Shopify/shopify-app-bridge#266]
### Describe the feature you’d like

The image uploader and selector model being used on various parts of the admin (product image, theme images etc) is pretty robust. It would be great if that could be exposed to the App Bridge api too, similar to the product picker api. We then won’t have to make that ourselves. Similar to product picker api, apps with files api access scope could make use of that modal.

### Describe alternatives you've considered
We just upload files directly to the files section of Shopify. In many cases, merchants might already have the image uploaded to Shopify somewhere else and so, it may result in unnecessary duplicate files on Shopify's end. Shopify could also save storage costs by giving us this api 🙂

### Additional context
The native image picker is just good and works flawlessly. Would love to let our merchants use that on our apps too.

**Comment by darrynten** [comment]
Seconded!

**Comment by Stef733** [comment]
I would also use it in my app immediately if it was made available.

**Comment by jzazove** [comment]
This a great feature request.  We can look into supporting this in the future.

**Comment by kelreel** [comment]
It would be very helpful, nice request

**Comment by gonzalomelov** [comment]
+1

**Comment by jkentjnr** [comment]
Kind of mandatory ... a lot of people writing average / bad implementations of this is bad for business, Shopify.

**Comment by KaziMuktadir-BS23** [comment]
Yeah, native file picker of shopify would be so much better than what we currently have

**Comment by thibaultmthh** [comment]
+1 waiting for an update

**Comment by gsouf** [comment]
Looking for it as well, hope it's coming in 2026!

**Comment by Zardw** [comment]
+10086

**Comment by charlesdobson** [comment]
Hi everyone!

We've added the `pick:shopify/File` intent to the Intents API so you can open the existing file picker modal right from your app. Check out the [changelog post](https://shopify.dev/changelog/intents-api-file-picker) and the [dev docs](https://shopify.dev/docs/api/app-home/apis/user-interface-and-interactions/intents-api?example=pick-files) and let us know if you have any questions!
